src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -357,47 +369,159 @@ def executeMcpTestSteps(self, config, test_type):
+                 fss_high_freq <= interference.FSS_TTC_HIGH_FREQ_HZ and 
+                 fss_ttc_flag is True):
+          # Call iap reference model for FSS blocking
+          fss_blocking_ap_iap_ref_values = iap.\
+              performIapForFssBlocking(fss_record, sas_uut, sas_th_fad_objects)
+          # Call aggregate interference model for FSS blocking
+          fss_blocking_aggr_interference = aggregate_interference.\
+              calculateAggregateInterferenceForFssBlocking(fss_record, cbsd_list)
+          # Compare the interference values calculated from both models
+          self.compareIapAndAggregateResults(fss_blocking_ap_iap_ref_values,\
+              fss_blocking_aggr_interference, 'point')
+
+    # Calculate and compare the interference value for ESC protection point 
+    if 'escRecords' in protection_entities:
+      for esc_record in protection_entities['escRecords']:
+        # Call iap reference model for esc
+        esc_ap_iap_ref_values = iap.\
+            performIapForEsc(esc_record, sas_uut, sas_th_fad_objects)
+        # Call aggregate interference model for esc 
+        esc_aggr_interference = aggregate_interference.\
+            calculateAggregateInterferenceForEsc(esc_record, cbsd_list)
+        # Compare the interference values calculated from both models
+        self.compareIapAndAggregateResults(esc_ap_iap_ref_values, esc_aggr_interference, 'point')
+      
+  def compareIapAndAggregateResults(self, ap_iap_ref_values, aggr_interference, entity_type):
+    """Verify aggregate interference is less than or equal to ap_iap_ref value calculated 
+    by IAP model plus a delta for each FSS and ESC sensor protection point, and for at
+    least 95% of the protection points of each PPA and GWPZ.
 
-       Returns:
-         Aggregate Interference list
+    Args:
+      ap_iap_ref_values : Aggregate interference output calculated from IAP 
+                             reference model 
+      aggr_interference : Aggregate interference output calculated from Aggregate
+                          Inteference reference model 
+      entity_type : To identify its a point protection or area protection entity 
+
+    Returns:
+       Pass or Fail based on the comparison of the values within range
     """
-    pass
+    iter_count = 0 # Variable to count the number of interference entries
+    match_count = 0 # Variable to count the number of matching interference entries
+    
+    for lat_dict in ap_iap_ref_values:
+      long_dict = ap_iap_ref_values[lat_dict]
+      long_dict_aggr = aggr_interference[lat_dict]
+      for interf in long_dict:
+        index = 0 
+        for interf_iap in long_dict[interf]:
+          # The aggregate interference should be less than 
+          # or equal to the iap interference + iap margin threshold
+          if long_dict_aggr[interf][index] <= \
+                interf_iap+interference.dbToLinear(IAP_MARGIN):
+            match_count+=1
+            index+=1
```

Comment:
Are you sure the index is within the the `if` ??
The whole condition could be made more pythonic as following:
```
for interf_iap, aggr in zip(long_dict[interf], long_dict_aggr[interf]):
   iter_cnt += 1
   if  aggr <= interf_iap * iap_margin_lin:
      match_cnt += 1

```

Reply:
changed   as per comment